base/file.jl

```diff
@@ -517,20 +517,27 @@ function mktemp(parent::AbstractString=tempdir(); cleanup::Bool=true)
     return (filename, Base.open(filename, "r+"))
 end
 
+# generate a random string from random bytes
+function _rand_string()
+	nchars = 10
+	A = Vector{UInt8}(undef, nchars)
+	ccall((:SystemFunction036, :Advapi32), stdcall, UInt8, (Ptr{Cvoid}, UInt32), A, sizeof(A))
```

Comment:
This function can fail. The return value should be checked, but there's not much we can do to recover to just throw an error.

Reply:
Actually, we do the same thing in the stdlib, so I'm gonna merge this and then fix them both at once.